src/build.py

```diff
@@ -74,7 +74,7 @@ def omit_includes(str, files):
     return str
 
 def fix_comments(str):
-    return re.sub(r"//(.*)(\n|$)", "/* \\1 */\\2", str)
+    return re.sub(r"//(.*)(" + os.linesep + r"|$)", "/* \\1 */\\2", str)
```

Comment:
This will, for some bizzare reason, always give priority to \n. This results in each comment being replaced with ```\r /* comment */ \n```

Reply:
Apparently I need to learn Python regexes better :cry:. I think the issue is with the `$` character, but I think the easiest thing would be to rewrite it to simply `re.sub( "//([^\r\n]*)\r?\n?", "/* \\1 */\n", s )`. Does this work with python 2.x on your system?

Also if we're already at it, we should discourage the use of `str` as variable label as it's a keyword in Python.

Reply:
The problem with that one is that it appends LF instead of the system-specific newline. I think it might be best to just completely remove the comment replacing feature, as `//` is fully supported in C99 and later, and in C90 if strict mode is not used.

Reply:
>The problem with that one is that it appends LF instead of the system-specific newline.

That was my goal :wink:.

>I think it might be best to just completely remove the comment replacing feature, as `//` is fully supported in C99 and later, and in C90 if strict mode is not used.

We're trying to be strictly C89 compatible for good reasons (and this presumably won't change during the coming few years). That's why we introduced the "replacing feature".

Reply:
But of course, I could write the replacement feature in POSIX shell and assume each Windows dev has [git for windows](https://git-scm.com/download/win ) installed allowing them to run `paq.sh` and remove `paq.bat` completely. But I'd prefer leaving it up to the `build.py` script.

Reply:
> That was my goal 😉.

See my reply below as to why this isn't recommended.

> But of course, I could write the replacement feature in POSIX shell and assume each Windows dev has git for windows installed allowing them to run paq.sh and remove paq.bat completely. But I'd prefer leaving it up to the build.py script.

I'm not sure if this would be a good assumption to make. Somehow we need to fix the regex so that it searches for the line-endings in the order that they are specified, instead of always giving precedence to `\n`.